### PR TITLE
URGENT bugfix: Path is allowed for DiskType Mount, not Root

### PR DESCRIPTION
**Pull request by timcharper**
This fixes major issue for using mount disk resources with Marathon.

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/state/VolumeTest.scala [hunk 1]
```diff
@@ -74,4 +75,26 @@ class VolumeTest extends MarathonSpec with Matchers {
...
+    DiskSource(DiskType.Root, None).asMesos shouldBe None
+    val Some(pathDisk) = DiskSource(DiskType.Path, Some("/path/to/folder")).asMesos
+    pathDisk.getPath.getRoot shouldBe "/path/to/folder"
+    pathDisk.getType shouldBe Source.Type.PATH
+
+    val Some(mountDisk) = DiskSource(DiskType.Mount, Some("/path/to/mount")).asMesos
+    mountDisk.getMount.getRoot shouldBe "/path/to/mount"
+    mountDisk.getType shouldBe Source.Type.MOUNT
+
+    a[java.lang.IllegalArgumentException] shouldBe thrownBy {
```
Comment (jasongilanfarr) [inline src/test/scala/mesosphere/marathon/state/VolumeTest.scala]: Why fully qualified?

Comment (jdef) [inline src/test/scala/mesosphere/marathon/state/VolumeTest.scala]: i don't think qualification matters here. the fact that there's a Some(path) at all is enough

**Comment by timcharper** [comment]
I found another issue and pushed another patch on this pull request.

File: src/test/scala/mesosphere/marathon/state/VolumeTest.scala (hunk 1 above)
Comment (jasongilanfarr) [inline src/test/scala/mesosphere/marathon/state/VolumeTest.scala]: I was referring to prefixing IllegalArgumentException.

**Comment by jasongilanfarr** [comment]
@timcharpher - just want to make sure you didn't miss adding the additional commit

**Comment by timcharper** [comment]
@jasongilanfarr thanks :) would you like me to rebase and squash this patch?

**Comment by jasongilanfarr** [comment]
That'd be great

**Comment by timcharper** [comment]
Squashed

**Comment by jasongilanfarr** [comment]
Ok to test

**Pull request merged**